src/Symfony/Component/Notifier/Bridge/OneSignal/OneSignalTransport.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\OneSignal;
+
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\PushMessage;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Tomas Norkūnas <norkunas.tom@gmail.com>
+ */
+final class OneSignalTransport extends AbstractTransport
+{
+    protected const HOST = 'onesignal.com';
+
+    private $appId;
+    private $apiKey;
+
+    public function __construct(string $appId, string $apiKey, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->appId = $appId;
+        $this->apiKey = $apiKey;
+        $this->client = $client;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('onesignal://%s?app_id=%s', $this->getEndpoint(), urlencode($this->appId));
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof PushMessage && (null === $message->getOptions() || $message->getOptions() instanceof OneSignalOptions);
```

Comment:
In other bridges we do not check the options type in the supports method. We do this only in the `doSend` method and throw an exception

Reply:
I've looked to slack transport as an example, so did same way

Reply:
Will have a look, thanks